assets/js/googlesitekit/datastore/user/user-input-settings.js

```diff
@@ -0,0 +1,202 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+import isPlainObject from 'lodash/isPlainObject';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from './constants';
+import { createFetchStore } from '../../data/create-fetch-store';
+const { commonActions, createRegistrySelector } = Data;
+
+function fetchStoreReducerCallback( state, inputSettings ) {
+	return {
+		...state,
+		inputSettings: Object
+			.keys( inputSettings )
+			.reduce( ( accum, key ) => ( { ...accum, [ key ]: inputSettings[ key ].values } ), {} ),
+	};
+}
+
+const fetchGetUserInputSettingsStore = createFetchStore( {
+	baseName: 'getUserInputSettings',
+	controlCallback: () => API.get( 'core', 'user', 'user-input-settings', undefined, { useCache: false } ),
+	reducerCallback: fetchStoreReducerCallback,
+} );
+
+const fetchSaveUserInputSettingsStore = createFetchStore( {
+	baseName: 'saveUserInputSettings',
+	controlCallback: ( settings ) => API.set( 'core', 'user', 'user-input-settings', { settings } ),
+	reducerCallback: fetchStoreReducerCallback,
+	argsToParams: ( settings ) => settings,
+	validateParams: ( settings ) => {
+		invariant( isPlainObject( settings ), 'valid settings are required.' );
+	},
+} );
+
+// Actions
+const SET_USER_INPUT_SETTINGS = 'SET_USER_INPUT_SETTINGS';
+const SET_USER_INPUT_SETTING = 'SET_USER_INPUT_SETTING';
+
+const baseInitialState = {
+	inputSettings: undefined,
+};
+
+const baseActions = {
+	/**
+	 * Sets user input settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} values User input settings.
+	 * @return {Object} Redux-style action.
+	 */
+	setUserInputSettings( values ) {
+		return {
+			type: SET_USER_INPUT_SETTINGS,
+			payload: { values },
+		};
+	},
+
+	/**
+	 * Sets user input setting.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string}         settingID Setting key.
+	 * @param {Array.<string>} values    User input settings.
+	 * @return {Object} Redux-style action.
+	 */
+	setUserInputSetting( settingID, values ) {
+		return {
+			type: SET_USER_INPUT_SETTING,
+			payload: {
+				settingID,
+				values,
+			},
+		};
+	},
+
+	/**
+	 * Saves user input settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {Object} Object with `response` and `error`.
+	 */
+	*saveUserInputSettings() {
+		const registry = yield Data.commonActions.getRegistry();
+		const settings = registry.select( STORE_NAME ).getUserInputSettings();
+
+		const { response, error } = yield fetchSaveUserInputSettingsStore.actions.fetchSaveUserInputSettings( settings );
+		if ( error ) {
+			// Store error manually since saveUserInputSettings signature differs from fetchSaveUserInputSettings.
+			registry.dispatch( STORE_NAME ).receiveError( error, 'saveUserInputSettings' );
```

Comment:
This is somewhat incorrect and would actually need to be like this:
```suggestion
			registry.dispatch( STORE_NAME ).receiveError( error, 'saveUserInputSettings', [] );
```

However, this is an unrelated problem and will be fixed via #2210, so we can ignore it here.